### ISSUE TITLE
Lazyload and image crawling issues

### ISSUE THREAD

**Issue by DimLeon** [verlok/lazyload#418]
There seems to be some image indexing issues as we noticed a great downfall in our website's images clicks/impressions. The screenshot is from google's search console and it displays our website's image performance in google's search results:

![image](https://user-images.githubusercontent.com/28803260/75567001-6ae1e600-5a59-11ea-93d4-bec19140b1c0.png).

The weird thing is that we have been using the lazyload plugin since April 2019, but as you can see in the above screenshot the downfall began occuring at about mid December and quite suddenly.
At this point now, if i google search our website and search for images there is not even one single photo appearing in results.

I have the vanilla-lazyload script (12.4.0) placed before `</body>.`
Just before this i have the I/O polyfill script.
I use the `data-scrset` attribute as well as `data-sizes` and have the `src` attribute blank (i don't use any low quality placeholder images).
 
I ran tests with puppeteer script (from puppeteer examples page: lazyimages_without_scroll_events.js), testing some of our website urls as well as some examples from the verlok/lazyload demos. This script checks if page is using scroll events to lazy load images. This plugin DOES NOT use scroll events, however all test were "Failed".

I also temporarily added a chrome extension (Website SEO checker) and tested some of my urls with plenty of lazy loaded images and result was: "No images found". 

I am not quite sure if the lazy loading technique is actually good for seo and proper crawling, however we have also created image sitemaps to help bots with indexing.

Of course i am not pointing this as a bug of this plugin, but i am just wondering if there is any workaround for websites that rely quite much on their good quality images. I was thinking to add all images inside <noscript> but not sure if it such a good idea as this could lead to heavy html files.
Any ideas?

**Comment by verlok** [comment]
I'm not a SEO expert so I'm not sure I can help.

First question I ask you is: how can you be sure that the drop in your google search console is due to the implementation of vanilla-lazyload? I mean, it could be for any other reason, can't it? Or the drop in the graphic matches the date on which you deployed a new version of your website using lazyload?

**Comment by verlok** [comment]
I think it might be interesting for you to read the final part of the discussion following issue #277. In the end of it, my understanding is that LazyLoad is 100% SEO friendly.

**Comment by verlok** [comment]
A couple of advices: 

- Try upgrading to 12.5.1 - Just to make sure you have the latest version
- Try removing the IntersectionObserver polyfill - It's not necessary! If you don't use it, all images will be loaded immediately on browsers that don't support it, meaning Internet Explorer 11 (almost unused in 2020), and you will save time and computing on all other browsers

Leave it like that for a couple of days and see how it goes on Google Search console.

**Comment by verlok** [comment]
If the above steps don't solve your issue, I would suggest you to use a hybrid approach to lazy loading, with hybrid meaning half native (which is supported [only in chromium-based browsers](https://caniuse.com/#feat=loading-lazy-attr)), half js-driven (for Safari, Firefox and others).

I wrote an article about it on Smashing Magazine --> [Hybrid Lazy Loading: A Progressive Migration To Native Lazy Loading
](https://www.smashingmagazine.com/2019/05/hybrid-lazy-loading-progressive-migration-native/).

The only downside of native lazy loading is that you wouldn't have any control on the loading / loaded events, no loading-related classes and callbacks.

Don't forget to let me (and the hundred people following this repo) know how it goes in the end!

**Comment by verlok** [comment]
> I added the polyfill just a week ago in order to cover case that bot doesn't support I/O.

Ok thanks. That wouldn't be necessary since if LazyLoad detects that I/O is not supported, it loads all the images immediately. As you can see [at this line in the source](https://github.com/verlok/lazyload/blob/master/src/lazyload.js#L23), it's the first check it does.

> I will remove the polyfill and upgrade to latest version and check again and of course I will update this thread with new results.

If you added the polyfill just a week ago, it's OK to remove it but we already know it's not the cause of the problem. 
But try and update to the latest version because Babel was adding code that wasn't compatible with all browsers.

It that doesn't solve, try migrating to native lazy loading. All at once, by removing the LazyLoad script, or progressively with the `use_native` option, as I explained in the above-mentioned [article on Smashing Mag](https://www.smashingmagazine.com/2019/05/hybrid-lazy-loading-progressive-migration-native/).

**Comment by verlok** [comment]
Hey @DimLeon ,
how is this going? Do you have any news?

**Comment by verlok** [comment]
Hey @DimLeon ,
please send an update  on what is going on, otherwise I’ll close this assuming that you solved.

**Comment by verlok** [comment]
Closing this. 
Feel free to reopen and send updates anytime.

**Comment by DimLeon** [comment]
It seems that all our images are now being crawled properly.  And the vanilla-lazyload had no direct relation to this.
Throughout all pages of the website there was a complicated menu with menu items and submenus containing extended html code, many images as well as some ajax calls to fetch content. For some reason, as soon as we removed all these ajax calls as well as some hundreds lines of html content, we noticed vast increase of image impressions in google search console.

No actual explanation on this. 
We had no errors, images were loading properly in the front, website speed was quite descent. 
However, the 95% of the website images were not crawled. 
And everything fixed when we removed some extended html code and 3 ajax functions.

So, nothing to do with LazyLoad script. Thanks you for support.
